internal/service/enrollmentrequest.go

```diff
@@ -233,29 +232,28 @@ func (h *ServiceHandler) ApproveEnrollmentRequest(ctx context.Context, request s
 	case nil:
 	}
 
+	// if the enrollment request was already approved we should not try to approve it one more time
 	if request.Body.Approved {
-
+		if v1alpha1.IsStatusConditionTrue(enrollmentReq.Status.Conditions, v1alpha1.EnrollmentRequestApproved) {
+			return server.ApproveEnrollmentRequest400JSONResponse{Message: "Enrollment request is already approved"}, nil
+		}
 		if request.Body.ApprovedAt != nil {
 			return server.ApproveEnrollmentRequest422JSONResponse{Message: "ApprovedAt is not allowed to be set when approving enrollment requests"}, nil
```

Comment:
This is the only API that returns 422 - can you please change all instances to 400 while you're at it?

Reply:
sure